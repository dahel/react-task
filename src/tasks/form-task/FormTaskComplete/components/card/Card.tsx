import { ReactNode } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  padding: 20px 10px;
  flex-wrap: wrap;
  justify-content: space-between;
  border: solid 1px #bdbdbd;
  position: relative;
  margin-top: 30px;
  opacity: ${(props: { pending: boolean }) => (props.pending ? 0.5 : 1)};
  pointer-events: ${(props: { pending: boolean }) => (props.pending ? 'none' : 'initial')};
  transition: all 0.2s;
`;

const Title = styled.h3`
  position: absolute;
  top: -28px;
  left: 10px;
  padding: 0 5px;
  background-color: white;
`;

interface IProps {
  children: ReactNode[];
  title: string;
  pending?: boolean;
}

const Card = ({ children, title, pending = false }: IProps) => {
  return (
    <Wrapper pending={pending}>
      <Title>{title}</Title>
      {children}
    </Wrapper>
  );
};

export default Card;
