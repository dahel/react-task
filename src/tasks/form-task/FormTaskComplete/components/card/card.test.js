import renderer from 'react-test-renderer';
import Card from './Card';

const props = {
  title: 'title',
};

describe('Card component', () => {
  test('renders properly for only-required props passed', () => {
    const component = renderer.create(<Card {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test(`renders properly in case 'pending' is true`, () => {
    const component = renderer.create(<Card {...props} pending={true} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test(`renders properly in case children components are defined`, () => {
    const component = renderer.create(
      <Card {...props}>
        <div>this is div</div>
      </Card>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
