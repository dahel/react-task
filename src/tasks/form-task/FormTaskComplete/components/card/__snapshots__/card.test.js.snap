// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders properly for only-required props passed 1`] = `
<div
  className="sc-bdvvtL cZVHgs"
>
  <h3
    className="sc-gsDKAQ iQGclo"
  >
    title
  </h3>
</div>
`;

exports[`Card component renders properly in case 'pending' is true 1`] = `
<div
  className="sc-bdvvtL grriIm"
>
  <h3
    className="sc-gsDKAQ iQGclo"
  >
    title
  </h3>
</div>
`;

exports[`Card component renders properly in case children components are defined 1`] = `
<div
  className="sc-bdvvtL cZVHgs"
>
  <h3
    className="sc-gsDKAQ iQGclo"
  >
    title
  </h3>
  <div>
    this is div
  </div>
</div>
`;
